### PR TITLE
NMS repackaging

### PR DESCRIPTION
**Pull request by Proximyst**
This PR blocks master.

Big thanks to @jpenilla for his work on the initial update.

### CODE REVIEW

**Comment by MiniDigger** [comment]
needs `./paper clean` to try it out

**Review by MiniDigger — APPROVED** [review]
seems fine. if its broken, we blame md5

File: scripts/decompile.sh
Comment (jpenilla) [inline scripts/decompile.sh]:
```suggestion
    jar xf "$decompiledir/$minecraftversion-mapped.jar" net/minecraft com/mojang/math
```

**Review by jpenilla — CHANGES_REQUESTED** [review]
We also need to extract the classes in `com.mojang.math`. importmcdev probably needs changes as well to work with `com.mojang.math` classes when/if we want to patch them.

File: Spigot-Server-Patches/0534-Fix-SPIGOT-5989.patch
Comment (jpenilla) [inline Spigot-Server-Patches/0534-Fix-SPIGOT-5989.patch]: t

**Review by jpenilla — CHANGES_REQUESTED** [review]

**Comment by theminecoder** [comment]
@Proximyst for people out of the loop, what is this very large pr doing

**Comment by Proximyst** [comment]
upstream update

**Comment by MiniDigger** [comment]
to iterate on that, upstream changed all nms mappings so have mojang mapped package names at dev time. this is most likely a preparation to also remap all classes, methods and fields to mojmap eventually. (nothing at runtime is changing, nothing for plugins is chaning, this is only for server dev time)

splitting up nms classes in multiple packages is challenging for us, since we rely on package private stuff all over the place, but its a change we are happy about, since it makes our mojmap migration easier too. we also need to touch all scripts to make them work properly with multiple packages.

**Review by zachbr — APPROVED** [review]
What a fun diff. Reviews a little better in git directly but not much.

I have confirmed this works nicely in Ubuntu 20.04 and Windows 10 in git bash after clearing the work dir (`/paper clean`). Jar layout looks as I'd expect and plugins look to be working correctly as expected.

Thanks all. Please ping a core team member immediately before merge so we can make sure the CI gets its workspace cleaned once.

**Pull request merged**
